runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -2985,18 +2985,32 @@ static void init_amHandling(void);
 
 static
 void init_ofiForAms(void) {
-  //
-  // Compute the amount of space we should allow for AM landing zones.
-  // We should have enough that we needn't re-post the multi-receive
-  // buffer more often than, say, every tenth of a second.  We know from
-  // the Chapel performance/comm/low-level/many-to-one test that the
-  // comm=ugni AM handler can handle just over 150k "fast" AM requests
-  // in 0.1 sec.  Assuming an average AM request size of 256 bytes, a 40
-  // MiB buffer is enough to give us the desired 0.1 sec lifetime before
-  // it needs renewing.  We actually then split this in half and create
-  // 2 half-sized buffers (see below), so reflect that here also.
-  //
-  const size_t amLZSize = ((size_t) 40 << 20) / 2;
+
+  // Specify the amount of space we should allocate for AM landing zones. We
+  // should have enough that we needn't re-post the multi-receive buffer more
+  // often than, say, every tenth of a second. The original value was 40MB as
+  // explained in the comment below. The default is now 64MB and it's
+  // configurable, although more experiments are necessary to perform a
+  // similar size computation on more modern hardware.
+  //
+  // Original comment:
+  // We know from the Chapel performance/comm/low-level/many-to-one test that
+  // the comm=ugni AM handler can handle just over 150k "fast" AM requests in
+  // 0.1 sec.  Assuming an average AM request size of 256 bytes, a 40 MiB
+  // buffer is enough to give us the desired 0.1 sec lifetime before it needs
+  // renewing.  We actually then split this in half and create 2 half-sized
+  // buffers (see below), so reflect that here also.
```

Comment:
In general I don't love journal style comments in code, but guess I don't want to lose the historical context for how this value was chosen either. I guess ok for now, but would be nice to have a simpler explanation for the value.

Reply:
I prefer them in the code because otherwise the reader may not realize they exist, much less where to find them. And including a URL or reference to a github issue seems more fragile. I don't want to lose this context until we have a chance to experiment with the default size.